### ISSUE TITLE
python: move, remove and copy behaviour seems to be modified

### ISSUE THREAD

**Issue by kpsherva** [xrootd/xrootd#2174]
We have noticed that between 5.5.5 and 5.6.x versions there are different responses on the file operations (move, remove, copy) 
please check for details in (5.5.5 tests are passing while 5.6.x are failing on the mentioned operations)
https://github.com/inveniosoftware/xrootdpyfs/actions/runs/7591815790/job/20680387171

Please let me know via Mattermost if there are more details needed

**Comment by amadio** [comment]
What I see is [this error](
https://github.com/inveniosoftware/xrootdpyfs/actions/runs/7591815790/job/20680387171#step:4:257) may be due to [this snippet](https://github.com/inveniosoftware/xrootdpyfs/blob/master/xrootdpyfs/fs.py#L734-L736) which tried to remove the destination directory if it exists, you are getting an error saying the directory being removed is not empty. I don't think there is a bug in XRootD in this case at least.

**Comment by kpsherva** [comment]
> What I see is [this error](https://github.com/inveniosoftware/xrootdpyfs/actions/runs/7591815790/job/20680387171#step:4:257) may be due to [this snippet](https://github.com/inveniosoftware/xrootdpyfs/blob/master/xrootdpyfs/fs.py#L734-L736) which tries to remove the destination directory if it exists. You are getting an error saying the directory being removed is not empty, which is a valid response by the server. I don't think there is a bug in XRootD in this case at least. I mean, if you want to write `file.txt` into `/data/file.txt` by calling `copy('file.txt', '/data/', overwrite=True)` (see arguments to the call `fs.copy(src_exists, dst_folder_exists, overwrite=True)` in the error logs), you probably do not want to delete `/data/`, which is what I think is happening here.

Thank you for the explanation. Ideally we would like to have an option of deleting a folder which is emptied by fe. moving a file. I am not sure however why in this case the 5.5.x client's tests are passing, since the operations tested are exactly the same? This is the reason which made me think there is incompatible behaviour of file operations between v5.5.x and 5.6.x. I am concerned this is implicit from our point of view and will have a nagative impact on instances upgraded to v5.6.x
Is there any way we could make sure the behaviour is consistent between versions?

**Comment by amadio** [comment]
I don't think that we can accommodate this request. Looking a bit further into this, I see that your issue is very likely caused by 3715a81a5807a2fd2aee5dbbcd2aac1a74d5acc6 and 6f591adbbe9108650b9f7fec2e1776c4152c1d49, which were put in place in #2016 to fix #1896. You will likely need to adapt your code (for example, [here](https://github.com/inveniosoftware/xrootdpyfs/blob/9cf4bb900b2773cef61c27dfe9c4c274ef20d0e4/xrootdpyfs/fs.py#L408)) such that it knows that both 3005 and 3018 mean the directory is not empty. Also, note that if you recursively remove `/data/` when copying `/data/file.txt` with overwrite=True, your code now would try remove `/data/otherfile.txt` which is likely an unwanted side effect.
